.pre-commit-config.yaml

```diff
@@ -0,0 +1,53 @@
+default_language_version:
+  python: python3
+
+ci:
+  autofix_prs: true
+  autoupdate_commit_msg: "[pre-commit.ci] pre-commit suggestions"
+  autoupdate_schedule: monthly
+  # submodules: true
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.5.0
+    hooks:
+      #  - id: end-of-file-fixer
+      #  - id: trailing-whitespace
+      - id: check-json
+      - id: check-yaml
+      - id: check-toml
+      #  - id: check-docstring-first
+      #  - id: check-executables-have-shebangs
+      #  - id: check-case-conflict
+      #  - id: check-added-large-files
+      - id: detect-private-key
+
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.3.3
+    hooks:
+      - id: ruff
+        name: Run Ruff (Flake8)
+        args: ["--fix"]
+
+  # todo: fix failing cases
```

Comment:
this seems to be failing silently for a long time so will be addressed in separate PR, see runs on master: https://github.com/awslabs/gluonts/actions/runs/8569986244/job/23486992584